portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java

```diff
@@ -0,0 +1,92 @@
+package com.google.sps.servlets;
+
+import com.google.sps.model.Comment;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Date;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for listing tasks. */
+@WebServlet("/list-comments")
+public class ListCommentsServlet extends HttpServlet {
+
+  private int requestCommentCount;
```

Comment:
the logic around request comment count is overly complicated and hard to understand. Why do you have a post request that only seems to define the request count and then a get request that produces the result? It is the get request that should tell the server how many comments are requested in which case there would be no need for class variable.

Reply:
Thank you for the debugging session! The Servlet requires only GET now. Previously, POST was a work around because <form> didn't work well with GET.